serde_derive/src/lib.rs

```diff
@@ -9,9 +9,455 @@
 //! # fn main() {}
 //! ```
 //!
-//! Please refer to [https://serde.rs/derive.html] for how to set this up.
+//! Please refer to [https://serde.rs/derive.html] for how to set this up and
+//! troubleshoot.
 //!
 //! [https://serde.rs/derive.html]: https://serde.rs/derive.html
+//!
+//! # Examples
+//!
+//! An example below while showing the differences between attributes.
+//!
+//! - [Container attributes](#container-attributes) — apply to a struct or enum declaration.
+//! - [Variant attributes](#variant-attributes) — apply to a variant of an enum.
+//! - [Field attributes](#field-attributes) — apply to one field in a struct or in an enum variant.
+//!
+//! ```
+//! # extern crate serde;
+//! use serde::{Serialize, Deserialize};
+//!
+//! #[derive(Serialize, Deserialize)]
```

Comment:
I would prefer not to start with this as the first code example. This code serves a very specific purpose in https://serde.rs/attributes.html and is not the front page of the book. The top example needs to be something that is as representative as possible of real world usage.